### PR TITLE
erase duplicate dependency, adding search functionalities.

### PR DESCRIPTION
**Pull request by matiastealdi**
Erase duplicate dependency.
Adding search functionalities. Search by description and search by signals. Prototype, not final version yet.

### CODE REVIEW

File: src/main/scala/com/teracode/beacons/storage/MemoryStorage.scala [hunk 1]
```diff
@@ -40,4 +40,12 @@ object LocationMemoryStorage extends CRUDOps[Location] {
   def search(): Future[Seq[Location]] = Future {
     Locations.values.toList
   }
+
```
Comment (matiastealdi) [inline src/main/scala/com/teracode/beacons/storage/MemoryStorage.scala:43]: Dummy functions to make MemoryStorage a trait of CRUDOPs

File: src/main/scala/com/teracode/beacons/routing/LocationService.scala [hunk 2]
```diff
@@ -50,46 +67,70 @@ trait LocationService extends HttpService {
...
   def searchRoute = get {
     path(LocationsPath) {
-      onSuccess(locationESStorage.search()) { result =>
+      onSuccess(storage.search()) { result =>
         complete(result)
       }
     }
   }
 
+
```
Comment (matiastealdi) [inline src/main/scala/com/teracode/beacons/routing/LocationService.scala]: These lines are here to make swagger recognize SignalSearch as a type. With this lines, swagger shows SignalSearch as a model with its schema.
It should be removed with the new version of swagger-core according to the issue in the commentary.

**Comment by lregnier** [comment]
Search HTTP interface should be GET instead of POST, as the request does not modify the state of the server (idempotent):

http://stackoverflow.com/questions/4088350/is-rest-delete-really-idempotent

Further suggestions to refactor the search parameters at the HTTP layer:

http://stackoverflow.com/questions/207477/restful-url-design-for-search

**Comment by lregnier** [comment]
Search operations could be separated into another trait from the CRUDOps.

**Comment by matiastealdi** [comment]
About "Search operations could be separated into another trait" I am agree. Although, we should have something like :   

```
trait SomeName extends CRUDOps with SearchOps
```

and than defines BaseLocationService as :  

```
class BaseLocationService(val storage: SomeName[Location])  
```

What do you thinks?

**Pull request merged**